src/plugins/intel_cpu/src/nodes/pooling.cpp

```diff
@@ -365,17 +365,52 @@ void Pooling::prepareParams() {
         DnnlDesriptor desc{desc_ptr};
         pooling_v2_forward::primitive_desc prim_desc;
         primitive_desc_iterator itpd = desc.createPrimitiveDescriptorIterator(engine, key.attr);
+
+        executorPtr execPtr = nullptr;
         while (static_cast<bool>(itpd)) {
             impl_desc_type impl_type = parse_impl_name(itpd.impl_info_str());
-
             if (impl_type == key.implType) {
-                prim_desc = itpd.get();
+                auto prim_desc = pooling_v2_forward::primitive_desc(itpd.get());
+                execPtr = std::make_shared<PoolingExecutor>(prim_desc,
+                                                            key.inp->getDnnlDesc(),
+                                                            key.out->getDnnlDesc(),
+                                                            engine);
                 break;
             }
             if (!itpd.next_impl())
-                return nullptr;
+                break;
         }
-        return std::make_shared<pooling_v2_forward>(prim_desc);
+
+        if (!execPtr) {
+            auto inDesc =  dnnl::memory::desc(DnnlExtensionUtils::convertToDnnlDims(key.inp->getShape().getStaticDims()),
+                                              key.inp->getDataType(),
+                                              memory::format_tag::any);
+            auto outDesc = dnnl::memory::desc(DnnlExtensionUtils::convertToDnnlDims(key.out->getShape().getStaticDims()),
+                                              key.out->getDataType(),
+                                              memory::format_tag::any);
+
+            auto desc_ptr = createDescriptorHelper(key.inp->getDnnlDesc(),
+                                                   key.out->getDnnlDesc(),
```

Comment:
The is a problem. `createDescriptorHelper` with the existing tensor layouts from the key, which do not guarantee the best possible performance. However, `inDesc` and `outDesc` are with the `any` layout specifically to allow oneDNN select the best possible implementation on the platform. So please replace the descriptors from the key with the descriptors `inDesc` and `outDesc` in the `createDescriptorHelper` parameters list.

Reply:
Pooling forward descriptor initialization doesn't allow to specify `any` layout for input descriptor. The case when we use `inDesc` from the `key` and specify `any` for `outDesc` shows worse performance than case when both descriptors are taken from `key`
So in this case it makes no sense to introduce PoolingExecutor - these changes were reverted. Please review again

Reply:
Got it. Thank you for clarification.